tests/kernel/update_kernel.pm

```diff
@@ -455,28 +460,32 @@ sub run {
         return;
     }
 
-    my $repo = get_var('KOTD_REPO');
+    my $repo = is_sle_micro('>=6.0') ? get_var('OS_TEST_REPOS') : get_var('KOTD_REPO');
     my $incident_id = undef;
 
     unless ($repo) {
         $repo = get_required_var('INCIDENT_REPO');
         $incident_id = get_required_var('INCIDENT_ID');
     }
 
+    install_internal_certificate if (is_sle_micro('>=6.0'));
```

Comment:
We should avoid this. Why is `OS_TEST_REPOS` set to HTTPS URL instead of plain HTTP one?

Reply:
@jlausuch could please comment it? Logic comes from `trancactiona/install_updates`.

Reply:
@mdoucha would be `$repo =~ s/https:/http:/g;` ok for you ?

Reply:
`$repo =~ s/^https:/http:/;` inside the `zypper ar` loop would be safer.

Reply:
Since the `zypper ar` is in two different places, it'd also make sense to create a function for it that'll take the repos string with repo name prefix and return the list of added repo names.

Reply:
I.e.

    my $repo_names = add_test_repos($repo, 'kernel-update');

Reply:
Function is good addition, but it can fit only to `first_azure_release`, `update_kernel` and `parse_incident_repo` for ULP. The loop for adding `kgraft-update` repository in `prepare_kgraft` function is doing much more than simply adding repositories.

Reply:
The only common pattern is  `my @repos = split(",", $repo);`.

Reply:
`prepare_kgraft` can do the extra steps by looping over the returned list of repo names.

Reply:
It means that we just need to add `s/^https:/http:/` to `prepare_kgraft` and not adding additional generic function.

Reply:
`update_kernel` is not used in live patching at all and I don't want to touch what is not needed

Reply:
```
=head2 add_test_repos

 add_test_repos(%args);

Add one or multiple test repositories separated by commas. The added repositories
contain a base name with numbered suffixes (name-1, name-2, etc.). It is possible
to replace https in the URL with http to avoid certificate issues.

C<%args> should have the following keys defined:
- C<repo>: one or multiple repositories separated by commas
- C<name>: base repository name
- C<replace_https>: replace https with http in the repository URL

=cut

sub add_test_repos {
    my ($repo, $name, $replace_https) = @_;

    my @repos = split(",", $repo);
    while (my ($i, $val) = each(@repos)) {
        $val =~ s/^https:/http:/ if $replace_https;
        zypper_call("ar $val $name-$i");
    }
```
Would be good for  first_azure_release, update_kernel and parse_incident_repo for ULP, but not kgraft

Reply:
We can only replace `my @repos = split(",", $repo);`  by new funcion, which just split repo and replaces https and returns array of repos without name.

Reply:
sub add_test_repos {
        my ($repo, $prefix, $replace_https) = @_;
    
        my @ret;
        my @repos = split(",", $repo);
        while (my ($i, $val) = each(@repos)) {
            $val =~ s/^https:/http:/ if $replace_https;
            my $name = "$prefix-$i";
            zypper_call("ar -G $val $name");
            push @ret, $name
        }
        return \@ret;
    }

Reply:
Also, you're using positional arguments in the function but you document them as keyword arguments in the proposal. Pick one or the other. There's a major difference in how they work.

Reply:
If you call the function according to the proposed documentation:

    add_test_repos(repo => 'http://foo/bar', name => 'kernel-update', replace_https => 1)

You'll get:
- `$repo == 'repo'`
- `$name == 'http://foo/bar'`
- `$replace_https == 'name'`

Reply:
Thanks  for pointing on it, it was just copy paste of quick unfinished draft.

Reply:
We could switch `OS_TEST_REPOS` to http if needed. Not sure if this will have any impact...

Reply:
As we agreed that we will be submitting jobs to openQA manually, we will use it  with http and when is tooling for automated submissions implemented, it can be set there as well.